pandas/plotting/_matplotlib/style.py

```diff
@@ -57,28 +57,8 @@ def random_color(column):
             raise ValueError("color_type must be either 'default' or 'random'")
 
     if isinstance(colors, str):
-        conv = matplotlib.colors.ColorConverter()
-
-        def _maybe_valid_colors(colors):
-            try:
-                [conv.to_rgba(c) for c in colors]
-                return True
-            except ValueError:
-                return False
-
-        # check whether the string can be convertible to single color
-        maybe_single_color = _maybe_valid_colors([colors])
-        # check whether each character can be convertible to colors
-        maybe_color_cycle = _maybe_valid_colors(list(colors))
-        if maybe_single_color and maybe_color_cycle and len(colors) > 1:
-            hex_color = [c["color"] for c in list(plt.rcParams["axes.prop_cycle"])]
-            colors = [hex_color[int(colors[1])]]
-        elif maybe_single_color:
+        if _is_single_color(colors):
```

Comment:
maybe put it in one line?
```python
`if isinstance(colors, str) and _is_single_color(colors)`
```
also nice to put an issue number below this line for future reference!

Reply:
Done